pex/cli/commands/lock.py

```diff
@@ -397,15 +656,48 @@ def _add_update_arguments(cls, update_parser):
                 )
             ),
         )
-        cls._add_lockfile_option(update_parser, verb="update")
-        cls._add_lock_options(update_parser)
-        cls.add_json_options(update_parser, entity="lock", include_switch=False)
-        cls._add_target_options(update_parser)
-        resolver_options_parser = cls._create_resolver_options_group(update_parser)
-        resolver_options.register_repos_options(resolver_options_parser)
-        resolver_options.register_network_options(resolver_options_parser)
-        resolver_options.register_max_jobs_option(resolver_options_parser)
-        resolver_options.register_use_pip_config(resolver_options_parser)
+
+    @classmethod
+    def _add_sync_arguments(cls, sync_parser):
+        # type: (_ActionsContainer) -> None
+        sync_parser.add_argument(
+            "--venv",
+            help=(
+                "Synchronize this venv to the contents of the lock after synchronizing the "
+                "contents of the lock to the passed requirements."
+            ),
+        )
+        sync_parser.add_argument(
+            "--venv-python",
+            help="Use this interpreter to create the `--venv` if the venv doesn't exist.",
+        )
+        sync_parser.add_argument(
+            "--pip",
+            "--retain-pip",
+            "--no-pip",
+            "--no-retain-pip",
+            action=HandleBoolAction,
+            default=False,
+            help=(
+                "When creating a venv that doesn't exist, install pip in it. When syncing an "
+                "existing venv, retain its pip, if any, even if pip is not present in the lock "
+                "file being synced. N.B.: When in `--no-retain-pip` mode (the default) only remove "
+                "pip if it is both present in the venv and not present in the lock file being "
+                "synced."
```

Comment:
If a lockfile has pip in it:

- does the `--no-retain-pip` (a) install pip to a new venv (or existing one without it), and (b) update it if an existing venv has a different version?
- does `--pip` use the version from the lockfile?

If so, the docs might be slightly clearer trying to convey something like:

- In the default `--no-retain-pip` mode, `pip` isn't special, and behaves like any other package in the lock (i.e. installed/updated/removed as required)
- With `--retain-pip`, `pip` is explicitly installed (aka, not removed) in venv, even if it is not in the lockfile (and if it is in the lockfile, the lockfile's version is used?)

Reply:
This is the grid that makes sense as I see it (keep in mind there is only a single bool here and the `--no-*` mean `False` / no Pip and the other two mean `True`, the venv should have Pip:
||Existing venv + resolve has Pip|Existing venv + resolve does not have Pip|No venv + resolve has Pip|No venv + resolve does not have Pip|
|---|---|---|---|---|
|`--pip`   |install resolve's Pip|`-m ensurepip`|install resolve's Pip|`-m ensurepip`|
|`--no-pip`|install resolve's Pip|uninstall Pip if installed|install resolve's Pip||

In other words the idea is `--no-pip` is ignored if the the resolve does in fact contain Pip. One tactic there would be to error instead: you told me no Pip but your resolve contains Pip. My thinking here is the main point is "sync"; so clearly the lock trumps. In a scenario where there are many venvs being synced, it makes sense to say `pex3 lock sync --no-pip ...` to ensure final venvs do not have pip unless in the lock (i.e.: a use case where you generally want the venv <-> lock mapping to be 1-1 / controlled by sync with no cowboying the venv by using Pip later ... unless the lock explicitly asks for Pip).

Reply:
I believe the 1 case in the grid not handled is `--pip` + "Existing venv + resolve does not have Pip". According to the grid, Pip should be installed via `-m ensurepip` but it is not. I'll fix that and add a test as well as take a second go at the option help docs.